test/unit/cross_lane_ops_test/device/dpp_ops.hpp

```diff
@@ -0,0 +1,306 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (C) 2021-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#ifndef ROCWMMA_DEVICE_DPP_OPS_HPP
+#define ROCWMMA_DEVICE_DPP_OPS_HPP
+
+#include "cross_lane_ops_util.hpp"
+
+namespace rocwmma
+{
+    ROCWMMA_DEVICE inline bool
+        isDppMasked(uint32_t id, uint32_t WriteRowMask, uint32_t WriteBankMask)
+    {
+        return (WriteRowMask & (1 << ((id >> 4) & 0x3)))
+               && (WriteBankMask & (1 << ((id >> 2) & 0x3)));
+    }
+
+    template <uint32_t SubgroupSize, uint32_t Element>
+    ROCWMMA_DEVICE inline uint32_t getDppBCastExpect(uint32_t input)
```

Comment:
Most if not all of the expect functions make assumption that inputs are sequential threadIdx.x values. Might be a good idea to disclaim that assumption here.

Reply:
defined a comment group of all ref generators of cross lane ops in test/unit/cross_lane_ops_test/device/cross_lane_ops_util.hpp

The assumption is added to the definition of the group. Included all these functions into it.